Assets/MixedRealityToolkit.SDK/Experimental/FollowSolver/Follow.cs

```diff
@@ -422,47 +476,108 @@ private bool AngularClamp(Vector3 refPosition, Quaternion refRotation, Vector3 c
             }
             else
             {
-                float angle = -AngleBetweenOnPlane(toTarget, currentRefForward, refRight);
-                float minMaxAngle = MaxViewVerticalDegrees * 0.5f;
-
-                if (angle < -minMaxAngle)
+                switch (angularClampMode)
                 {
-                    rotation = Quaternion.AngleAxis(-minMaxAngle - angle, refRight) * rotation;
-                    angularClamped = true;
-                }
-                else if (angle > minMaxAngle)
-                {
-                    rotation = Quaternion.AngleAxis(minMaxAngle - angle, refRight) * rotation;
-                    angularClamped = true;
+                    case AngularClampType.ViewDegrees:
+                    case AngularClampType.AngleStepping:
+                        {
+                            float angle = -AngleBetweenOnPlane(toTarget, currentRefForward, refRight);
+                            float minMaxAngle = MaxViewVerticalDegrees * 0.5f;
+
+                            if (angle < -minMaxAngle)
+                            {
+                                rotation = Quaternion.AngleAxis(-minMaxAngle - angle, refRight) * rotation;
+                                angularClamped = true;
+                            }
+                            else if (angle > minMaxAngle)
+                            {
+                                rotation = Quaternion.AngleAxis(minMaxAngle - angle, refRight) * rotation;
+                                angularClamped = true;
+                            }
+                        }
+                        break;
+
+                    case AngularClampType.RendererBounds:
+                    case AngularClampType.ColliderBounds:
+                        {
+                           Vector3 min = refRotation * new Vector3(0.0f, -extents.y, currentDistance);
+                           Vector3 max = refRotation * new Vector3(0.0f, extents.y, currentDistance);
+
+                            float minAngle = AngleBetweenOnAxis(toTarget, min, refRight);
+                            float maxAngle = AngleBetweenOnAxis(toTarget, max, refRight);
+
+                            if (minAngle < 0)
+                            {
+                                rotation = Quaternion.AngleAxis(minAngle, refRight) * rotation;
+                                angularClamped = true;
+                            }
+                            else if (maxAngle > 0)
+                            {
+                                rotation = Quaternion.AngleAxis(maxAngle, refRight) * rotation;
+                                angularClamped = true;
+                            }
```

Comment:
Can't we do this the same way as it's done above? There are issues with using AngleBetweenOnAxis, and we can avoid rotating and constructing new vectors and instead just measure the angle once. 

See the logic now under `AngularClampType.ViewDegrees` and `AngularClampType.AngleStepping`